File: 26.remove-duplicates-from-sorted-array.java
/*
 * @lc app=leetcode id=26 lang=java
 *
 * [26] Remove Duplicates from Sorted Array
 */

// @lc code=start
class Solution {
    public int removeDuplicates(int[] nums) {
        int slow = 1, fast = 1;
        while (fast != nums.length) {
            if (nums[fast] != nums[slow - 1]) {
                nums[slow] = nums[fast];
                slow += 1;
            }
            fast += 1;
        }
        return slow;
    }
}
// @lc code=end

File: 26.remove-duplicates-from-sorted-array.cpp
/*
 * @lc app=leetcode id=26 lang=cpp
 *
 * [26] Remove Duplicates from Sorted Array
 */

// @lc code=start

#include <vector>
using namespace std;

class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        auto rptr = nums.begin(), wptr = nums.begin();
        while (rptr != nums.end()) {
            if (rptr == nums.begin() || *rptr != *(wptr - 1)) {
                *wptr = *rptr;
                wptr += 1;
            }
            rptr += 1;
        }
        nums.erase(wptr, nums.end());
        return wptr - nums.begin();
    }
};
// @lc code=end
